test/runtests.jl

```diff
@@ -74,23 +74,27 @@ cd(@__DIR__) do
     #pretty print the information about gc and mem usage
     testgroupheader = "Test"
     workerheader = "(Worker)"
-    name_align    = maximum([length(testgroupheader) + length(" ") + length(workerheader); map(x -> length(x) + 3 + ndigits(nworkers()), tests)])
-    elapsed_align = length("Time (s)")
-    gc_align      = length("GC (s)")
-    percent_align = length("GC %")
-    alloc_align   = length("Alloc (MB)")
-    rss_align     = length("RSS (MB)")
+    name_align    = maximum([textwidth(testgroupheader) + textwidth(" ") + textwidth(workerheader); map(x -> textwidth(x) + 3 + ndigits(nworkers()), tests)])
+    elapsed_align = textwidth("Time (s)")
+    gc_align      = textwidth("GC (s)")
+    percent_align = textwidth("GC %")
+    alloc_align   = textwidth("Alloc (MB)")
+    rss_align     = textwidth("RSS (MB)")
     printstyled(testgroupheader, color=:white)
-    printstyled(lpad(workerheader, name_align - length(testgroupheader) + 1), " | ", color=:white)
+    printstyled(lpad(workerheader, name_align - textwidth(testgroupheader) + 1), " | ", color=:white)
     printstyled("Time (s) | GC (s) | GC % | Alloc (MB) | RSS (MB)\n", color=:white)
-    results=[]
-    print_lock = ReentrantLock()
+    results = []
+    print_lock = stdout isa Base.LibuvStream ? stdout.lock : ReentrantLock()
+    if stderr isa Base.LibuvStream && !islocked(stderr.lock)
+        stderr.lock = print_lock
```

Comment:
The file has single-threaded control at this point, so it doesn't really make a difference since it isn't locked. I'm not sure why I wrote the `islocked` check there—I can't really think of a reason to have it.

Reply:
Yeah I wasn't sure why this could happen. Maybe better as an `@assert !islocked(stderr.lock)` then or just remove it?

Reply:
Yeah, I just stripped it out